mkosi/config.py

```diff
@@ -3358,19 +3358,45 @@ def resolve_deps(images: Sequence[argparse.Namespace], include: Sequence[str]) -
     return sorted(images, key=lambda i: order.index(i.image))
 
 
-def parse_config(argv: Sequence[str] = (), *, resources: Path = Path("/")) -> tuple[Args, tuple[Config, ...]]:
+class ConfigAction(argparse.Action):
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = None
+    ) -> None:
+        assert option_string is not None
+
+        if values is None and self.nargs == "?":
+            values = self.const or "yes"
 
-    class ParseContext:
+        try:
+            s = SETTINGS_LOOKUP_BY_DEST[self.dest]
+        except KeyError:
+            die(f"Unknown setting {option_string}")
+
+        if values is None or isinstance(values, str):
+            setattr(namespace, s.dest, s.parse(values, getattr(namespace, self.dest, None)))
+        else:
+            for v in values:
+                assert isinstance(v, str)
+                setattr(namespace, s.dest, s.parse(v, getattr(namespace, self.dest, None)))
```

Comment:
Save `getattr(namespace, self.dest, None)` to a temp var?

Reply:
We have to re-evaluate it every time anyway so I don't think this gets us much